File: chess_cpp/validators/BishopMoveValidator.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "BishopMoveValidator.h"

bool BishopMoveValidator::validate(Board *board, Spot *start, Spot *end) {
    return abs(start->getX() - end->getX()) != 0 && abs(start->getX() - end->getX()) == abs(start->getY() - end->getY());
}

std::string BishopMoveValidator::getNotValidMessage() {
    return MessageHandler::getHellNoMessage();
}

File: oop_chess/src/chess/validators/BishopMoveValidator.java
package chess.validators;

import chess.Board;
import chess.Spot;
import chess.utils.MessageHandler;

public class BishopMoveValidator extends PieceMoveValidator {


    @Override
    public boolean validate(Board board, Spot start, Spot end) {
        return Math.abs(start.getX() - end.getX()) != 0 && Math.abs(start.getX() - end.getX()) == Math.abs(start.getY() - end.getY());
    }

    @Override
    protected String getNotValidMessage() {
        return MessageHandler.getHellNoMessage();
    }
}
